diff --git a/ext/wasm/api/pre-js.c-pp.js b/ext/wasm/api/pre-js.c-pp.js
index 8a4a0f9..b2c3147 100644
--- a/ext/wasm/api/pre-js.c-pp.js
+++ b/ext/wasm/api/pre-js.c-pp.js
@@ -46,7 +46,7 @@
      None of the above apply in ES6 builds, which uses a much simpler
      approach.
   */
-  Module['locateFile'] = function(path, prefix) {
+  Module['locateFile'] ??= function(path, prefix) {
 //#if target:es6-module
     return new URL(path, import.meta.url).href;
 //#else
