diff --git a/ext/wasm/api/pre-js.c-pp.js b/ext/wasm/api/pre-js.c-pp.js
index da314b9..b047b28 100644
--- a/ext/wasm/api/pre-js.c-pp.js
+++ b/ext/wasm/api/pre-js.c-pp.js
@@ -140,7 +140,9 @@
           ? "" : scriptDirectory)
     );
     sims.debugModule("instantiateWasm() uri =", uri, "sIMS =",this);
-    const wfetch = ()=>fetch(uri, {credentials: 'same-origin'});
+    const response = uri instanceof Response || uri instanceof Promise
+      ? uri
+      : fetch(uri, { credentials: 'same-origin' });
     const finalThen = (arg)=>{
       arg.imports = imports;
       sims.instantiateWasm = arg /* used by sqlite3-api-prologue.c-pp.js */;
@@ -149,10 +151,10 @@
     const loadWasm = WebAssembly.instantiateStreaming
           ? async ()=>
           WebAssembly
-          .instantiateStreaming(wfetch(), imports)
+          .instantiateStreaming(response, imports)
           .then(finalThen)
           : async ()=>// Safari < v15
-          wfetch()
+          Promise.resolve(response)
           .then(response => response.arrayBuffer())
           .then(bytes => WebAssembly.instantiate(bytes, imports))
           .then(finalThen)
